File: plotWavePoint.m
function h = plotWavePoint(ax, cond, label, varargin)
% PLOTWAVEPOINT  Overlay one wave condition on the base chart.
%
% Name-value options:
%   'MarkerSize' (default 24), 'FontSize' (default 10),
%   'FontName'   (default 'Times New Roman'),
%   'MarkerColor'(default []), 'ShowLabel' (default true)

    p = inputParser;
    addParameter(p,'MarkerSize',24);
    addParameter(p,'FontSize',10);
    addParameter(p,'FontName','Times New Roman');
    addParameter(p,'MarkerColor',[]);
    addParameter(p,'ShowLabel',true);
    parse(p,varargin{:});

    if nargin < 3, label = ''; end
    
    if ~isempty(p.Results.MarkerColor)
        col = p.Results.MarkerColor;
    elseif cond.broken
        col = [1 0 0];
    elseif strcmp(cond.waveType,'irregular')
        col = [0 .35 .8];
    else
        col = [0 .55 0];
    end
    
    if cond.broken
        mk = 'x'; sz = p.Results.MarkerSize+12;
    else
        mk = 'o'; sz = p.Results.MarkerSize;
    end
    
    hold(ax,'on');
    h = scatter(ax, cond.dL, cond.HL, sz, col, mk, 'filled', ...
                'LineWidth',1.5);
    if p.Results.ShowLabel && ~isempty(label)
        text(ax, cond.dL*1.08, cond.HL*1.05, label, ...
            'FontName',p.Results.FontName,'FontSize',p.Results.FontSize, ...
            'FontWeight','bold','Color',col);
    end
    hold(ax,'off');
end

File: plot_wave_point.py
import numpy as np


def plot_wave_point(ax, cond, label='', marker_size=24, font_size=11,
                    font_name='Times New Roman', marker_color=None,
                    show_label=True):
    if marker_color is not None:
        col = marker_color
    elif cond['broken']:
        col = (1, 0, 0)
    elif cond['waveType'] == 'irregular':
        col = (0, 0.35, 0.8)
    else:
        col = (0, 0.55, 0)

    if cond['broken']:
        mk = 'x'
        sz = marker_size + 12
    else:
        mk = 'o'
        sz = marker_size

    s = sz ** 2 / 4
    ax.scatter(cond['dL'], cond['HL'], s=s, c=[col], marker=mk,
               edgecolors='none', zorder=5)

    if show_label and label:
        ax.text(cond['dL'] * 1.08, cond['HL'] * 1.05, label,
                fontsize=font_size, fontname=font_name, fontweight='bold',
                color=col, zorder=6)
